src/System.Management.Automation/help/MUIFileSearcher.cs

```diff
@@ -304,12 +304,17 @@ private static Collection<String> NormalizeSearchPaths(string target, Collection
         /// </remarks>
         private static string GetMshDefaultInstallationPath()
         {
-            string returnValue = CommandDiscovery.GetShellPathFromRegistry(Utils.DefaultPowerShellShellID);
+            string returnValue = Utils.GetApplicationBase(Utils.DefaultPowerShellShellID);
 
+            //On CoreCLR we use ApplicationBase assembly location. That gives us the directory name.
+            //Hence we do not need to extract the directory name from returnValue.
+
+#if !CORECLR
             if (returnValue != null)
             {
                 returnValue = Path.GetDirectoryName(returnValue);
```

Comment:
GetMshDefaultInstallationPath had only one reference, replaced where it was called with ```Utils.GetApplicationBase(Utils.DefaultPowerShellShellID); ``` And deleted the function GetMshDefaultInstallationPath